src/main/java/org/kiwiproject/dropwizard/util/health/MonitoredJobHealthCheck.java

```diff
@@ -0,0 +1,143 @@
+import static java.util.Objects.isNull;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotNull;
+import static org.kiwiproject.base.KiwiStrings.f;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newResultBuilder;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newUnhealthyResult;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newUnhealthyResultBuilder;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.google.common.annotations.VisibleForTesting;
+import io.dropwizard.util.Duration;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.base.DefaultEnvironment;
+import org.kiwiproject.base.KiwiEnvironment;
+import org.kiwiproject.dropwizard.util.job.MonitoredJob;
+
+import java.time.Instant;
+
+@Slf4j
+@Getter(AccessLevel.PACKAGE) // For testing purposes
+public class MonitoredJobHealthCheck extends HealthCheck {
+
+    @VisibleForTesting
+    static final double DEFAULT_THRESHOLD_FACTOR = 2.0;
+
+    @VisibleForTesting
+    static final Duration MINIMUM_WARNING_THRESHOLD = Duration.minutes(1);
+
+    @VisibleForTesting
+    static final Duration DEFAULT_WARNING_DURATION = Duration.minutes(15);
+
+    private final MonitoredJob job;
+    private final Duration expectedFrequency;
+    private final Duration errorWarningDuration;
+    private final double thresholdFactor;
+    private final long lowerTimeBound;
+    private final KiwiEnvironment kiwiEnvironment;
+
+    @Builder
+    private MonitoredJobHealthCheck(MonitoredJob job,
+                                    Duration expectedFrequency,
+                                    Duration errorWarningDuration,
+                                    Double thresholdFactor,
+                                    Long lowerTimeBound,
+                                    KiwiEnvironment environment) {
+
+        this.job = requireNotNull(job, "job is required");
+        this.expectedFrequency = requireNotNull(expectedFrequency, "expectedFrequency is required");
+        this.errorWarningDuration = isNull(errorWarningDuration) ? DEFAULT_WARNING_DURATION : errorWarningDuration;
+        this.thresholdFactor = isNull(thresholdFactor) ? DEFAULT_THRESHOLD_FACTOR : thresholdFactor;
+        this.kiwiEnvironment = isNull(environment) ? new DefaultEnvironment() : environment;
+        this.lowerTimeBound = kiwiEnvironment.currentTimeMillis();
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        try {
+            var lastRun = job.getLastSuccess().get();
+            if (!job.isActive()) {
+                return buildHealthyResult(f("Job is inactive. (last run: {})", instantToStringOrNever(lastRun)));
+            }
+
+            var now = kiwiEnvironment.currentTimeMillis();
+            var lastFailure = job.getLastFailure().get();
+            if ((now - lastFailure) < errorWarningDuration.toMilliseconds()) {
```

Comment:
Rather than convert the `Duration` to milliseconds on every invocation, why not compute in the constructor and store the value?